networkx/algorithms/centrality/reaching.py

```diff
@@ -89,7 +89,9 @@ def global_reaching_centrality(G, weight=None, normalized=True):
     total_weight = G.size(weight=weight)
     if total_weight <= 0:
         raise nx.NetworkXError("Size of G must be positive")
-
+    # self loops
+    if total_weight > 0 and len(G) == 1:
+        raise nx.NetworkXPointlessConcept("G is a single node with self loop")
```

Comment:
I don't think this check is necessary - the exception will be hit when `local_reaching_centrality` is called. In principle it saves a bit of computation to raise here, but it's only the shortest path of a graph with a single node which should never be a bottleneck.

I'd vote to let `local_reaching_centrality` handle the raising, that way the exception message can be made more informative.

Reply:
Your are absolutely right. At first, I adopted your suggestion, but I thought of handling the raising in ``global_reaching_centrality`` as a way of **early exit**; however, I did not take the computational power needed for raise statement into consideration, so I think your approach is valid.